File: furama/Be2/src/main/java/com/example/demo/model/entity/Contract.java
package com.example.demo.model.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Contract {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "contract_id")
    private Long contractId;
    private String contractStartDate;
    private String contractEndDate;
    private Double contractDeposit;
    private Double contractTotalMoney;
    @ManyToOne
    @JoinColumn(name = "employee_id")
    private Employee employee;
    @ManyToOne
    @JoinColumn(name = "customer_id")
    private Customer customer;
    @ManyToOne
    @JoinColumn(name = "service_id")
    private EntityService services;
    @OneToMany(mappedBy = "contract")
    private Set<ContractDetail> contractDetails;
    private Boolean status;


}

File: furama/FE/src/app/model/contract.ts
import {Employee} from './employee';
import {Customer} from './customer';
import {Service} from './service';
import {ContractDetail} from './contract-detail';

export interface Contract {
  contractId:number;
  contractStartDate:string;
  contractEndDate:string;
  contractDeposit:number;
  contractTotalMoney:number;
  employee:Employee;
  customer:Customer;
  services:Service;
  contractDetail:ContractDetail[];
}
